tests/30rooms/70publicroomslist.pl

```diff
@@ -135,3 +135,105 @@
          Future->done( 1 );
       })
    };
+
+
+test "Can paginate public room list",
+   requires => [ local_user_fixture() ],
+
+   check => sub {
+      my ( $user ) = @_;
+
+      my $num_rooms;
+      my $next_batch;
+      my $prev_batch;
+
+      # A hash from room ID to number of times we saw the room.
+      my $counts = {};
```

Comment:
normally you'd use a regular hash rather than a hashref here:

```suggestion
      my %counts;
```

doesn't really matter though

Reply:
Originally I did that then everything went horribly wrong and it worked with a hashref and I still don't really understand perl :sob: